fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java

```diff
@@ -1973,6 +2016,42 @@ public long getConnectorSinkTargetMaxFileSize() {
     @VarAttr(name = ENABLE_PARTITION_COLUMN_VALUE_ONLY_OPTIMIZATION, flag = VariableMgr.INVISIBLE)
     private boolean enablePartitionColumnValueOnlyOptimization = true;
 
+    @VarAttr(name = ENABLE_USE_ANN)
```

Comment:
I see that many variables here are set in the rewriting rules. If these parameters do not require user settings, then it is not appropriate to put them here. If these configurations need to be confirmed during the rewriting process and sent to BE to execute, could you consider encapsulate them individually into  a separate configuration items?

Reply:
I got it.
I chose to do this earlier to make it easier to pass them through `TQueryOptions`, let me change it.

Reply:
@packy92 Hi, I've aggregated these configurations into `VectorSearchOptions`. I'm trying to rewrite these configurations by session, but I'm encountering two issues:

- The VariableMgr.save method throws an error: 'failed to write session variable: null'. I suspect this is due to VectorSearchOptions not being a primary type.

- The variable doesn't persist across multiple queries. For instance, in two separate runs, the value in VectorSearchOptions doesn't revert back to its original value."

Reply:
I changed `VectorSearchOptions` to Json String.
Can I use this method?